clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -874,6 +874,10 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
                             markCoordinatorUnknown();
                             future.raise(error);
                             return;
+                        } else if (error == Errors.FENCED_INSTANCE_ID) {
+                            log.error(fencedInstanceErrorMessage, groupInstanceId, this.getClass().getName());
```

Comment:
If the commit request is sent via async-commit, i.e. in `doCommitOffsetsAsync` then even if this error throws it would not cause the consumer to die, but only user's callback to see this error. Is that intentional?

Reply:
I added the exception throwing in `invokeCompletedOffsetCommitCallbacks`

Reply:
Cool.

Should we update the javadoc and the signature of `KafkaConsumer`'s corresponding APIs about the new throwable exceptions then?

Reply:
Sounds good!